test/e2e/kube_state_metrics_test.go

```diff
@@ -65,3 +66,115 @@ func TestKSMMetricsSuppression(t *testing.T) {
 	}
 
 }
+
+func TestKSMDenylistBoundsCheck(t *testing.T) {
+
+	// Update CMO config to include an invalid deny-list.
+	cfg, err := getOrCreateCMOConfig(t)
+	if err != nil {
+		t.Errorf("failed to get cluster-monitoring-config configmap: %v", err)
+	}
+	cfgCopy := cfg.DeepCopy()
+	invalidMetricExpression := "kube_lease_owner"
```

Comment:
not sure why `kube_lease_owner` is invalid, from api
`| metricDenylist | []string | Comma-separated list of metrics not to be enabled. This list comprises exact metric names and/or regex patterns. |`
seems it's valid, or it should be `^kube_lease_owner$`?

Reply:
Hey, @juzhao! The metric is invalid in the context of the metric deny-list test bounds. This basically implies that this metric cannot be disabled by the user (by specifying it in `metricDenylist`) due to the fact that it is not a part of the [default set of metrics that are deny-listed](https://github.com/openshift/cluster-monitoring-operator/blob/937046a650f32bdca4666cf6923feb956ed40d82/assets/kube-state-metrics/deployment.yaml#L38-L55), which is the only valid pool of valid metrics that can be denied by the user using this option, whose value will always be a subset of this pool.

Reply:
> seems it's valid since it can without regex patterns , or it should be ^kube_lease_owner$?

I noticed earlier today that the KSM deny-list compatibility took a bit of a hit for no reason in this patch, in the sense that the regexes specified by the user were earlier being matched against the payload lines, rather than a 1-1 against the metrics themselves.

So, earlier, the regexes still worked, but boundaries would cause unexpected results: `^foo$` would not match against a payload line that contained a `foo` metric, since the line did not start and end with just `foo`. To resolve this and maintain complete KSM compliance I have pushed a commit so users can specify expressions exactly as they would do in KSM's deny-list.

TL;DR `^kube_lease_owner$` is the correct way to specify an expression that matches against the `kube_lease_owner` metric **only**, although this will error since its not in the default deny-list set.

Reply:
thanks,
`[#2283](https://github.com/openshift/cluster-monitoring-operator/pull/2283) Expose KSM's `--metric-denylist` in `cluster-monitoring-config` ConfigMap, and allow users to enable metrics that are "denied" by default.`
I did not realized the `default` means [default set of metrics that are deny-listed](https://github.com/openshift/cluster-monitoring-operator/blob/937046a650f32bdca4666cf6923feb956ed40d82/assets/kube-state-metrics/deployment.yaml#L38-L55), the valid value for `metricDenylist` should be a subset of it, otherwise it's invalid